firefox-ios/Client/Frontend/TrackingProtection/TrackingProtectionButton.swift

```diff
@@ -0,0 +1,108 @@
+    private struct UX {
+        static let buttonCornerRadius: CGFloat = 12
+        static let buttonVerticalInset: CGFloat = 12
+        static let buttonHorizontalInset: CGFloat = 16
+        static let buttonFontSize: CGFloat = 16
+
+        static let contentInsets = NSDirectionalEdgeInsets(
+            top: buttonVerticalInset,
+            leading: buttonHorizontalInset,
+            bottom: buttonVerticalInset,
+            trailing: buttonHorizontalInset
+        )
+    }
+
+    private var viewModel = TrackingProtectionButtonViewModel()
+    private var backgroundColorNormal: UIColor = .yellow
+    private var foregroundColor: UIColor = .black
+    private var borderColor: UIColor = .black
+
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+
+        configuration = UIButton.Configuration.bordered()
+        titleLabel?.adjustsFontForContentSizeCategory = true
+        contentHorizontalAlignment = .leading
+    }
+
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    override func updateConfiguration() {
+        guard var updatedConfiguration = configuration else {
+            return
+        }
+
+        let transformer = UIConfigurationTextAttributesTransformer { [weak self] incoming in
+            var container = incoming
+            container.foregroundColor = self?.foregroundColor
+            container.backgroundColor = self?.backgroundColorNormal
+            container.font = FXFontStyles.Regular.body.scaledFont()
+            return container
+        }
+        updatedConfiguration.titleTextAttributesTransformer = transformer
+        updatedConfiguration.background.backgroundColor = backgroundColorNormal
+        updatedConfiguration.contentInsets = UX.contentInsets
+        updatedConfiguration.title = viewModel.title
+        updatedConfiguration.titleAlignment = .leading
+        updatedConfiguration.background.customView?.layer.borderWidth = 1
+        updatedConfiguration.background.customView?.layer.borderColor = borderColor.cgColor
+
+        configuration = updatedConfiguration
+    }
+
+    func configure(viewModel: TrackingProtectionButtonViewModel) {
+        guard var updatedConfiguration = configuration else {
+            return
+        }
+
+        self.viewModel = viewModel
+
+        updatedConfiguration.background.backgroundColor = backgroundColorNormal
+        updatedConfiguration.contentInsets = UX.contentInsets
+        updatedConfiguration.title = viewModel.title
+        updatedConfiguration.titleAlignment = .leading
+        updatedConfiguration.background.customView?.layer.borderWidth = 1
+        updatedConfiguration.background.customView?.layer.borderColor = borderColor.cgColor
+
+        // Using a nil backgroundColorTransformer will just make the background view
+        // use configuration.background.backgroundColor without any transformation
```

Comment:
not too clear on what this comment is referring to, can you further explain? thank you!

Reply:
if there were any pre-existing transformer applied to the background color, it can alter the background color based on certain states or conditions. By explicitly setting it to nil, we're ensuring a static background color that is defined in updatedConfiguration.background.backgroundColor

Reply:
Thanks for the explanation! what do you think about updating that comment so its more clear?